src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java

```diff
@@ -29,4 +44,42 @@ public CustomerDto getCustomer(final Long customerId) {
                 .orElseThrow(() -> new ResourceNotFoundException("Customer with id=" + customerId + " doesn't exist"));
     }
 
+    private final UserServiceImpl userService;
+
+    @Override
+    public Customer saveOrEditCustomer(CustomerDto dto) {
+        Optional<Customer> optionalCustomer = customerRepository.findById(dto.getId());
+        if (optionalCustomer.isPresent()) {
```

Comment:
you need to optimize this part, please take a look like done similar things in other services